src/main/java/graphql/schema/idl/SchemaPrinter.java

```diff
@@ -160,32 +156,50 @@ public String print(GraphQLSchema schema) {
         StringWriter sw = new StringWriter();
         PrintWriter out = new PrintWriter(sw);
 
+        try {
+            this.print(schema, out);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+
+        String result = sw.toString();
+        return result;
+    }
+
+    public void print(GraphQLSchema schema, Appendable appendable) throws IOException {
         GraphqlFieldVisibility visibility = schema.getFieldVisibility();
 
-        printer(schema.getClass()).print(out, schema, visibility);
+        Stream<TypePrinter.Appender> stream = printer(schema.getClass()).print(schema, visibility).map(Stream::of).orElse(Stream.empty());
 
         List<GraphQLType> typesAsList = schema.getAllTypesAsList()
                 .stream()
                 .sorted(Comparator.comparing(GraphQLType::getName))
                 .collect(toList());
 
-        printType(out, typesAsList, GraphQLInterfaceType.class, visibility);
-        printType(out, typesAsList, GraphQLUnionType.class, visibility);
-        printType(out, typesAsList, GraphQLObjectType.class, visibility);
-        printType(out, typesAsList, GraphQLEnumType.class, visibility);
-        printType(out, typesAsList, GraphQLScalarType.class, visibility);
-        printType(out, typesAsList, GraphQLInputObjectType.class, visibility);
-
-        String result = sw.toString();
-        if (result.endsWith("\n\n")) {
-            result = result.substring(0, result.length() - 1);
+        stream = Stream.concat(stream, Stream.of(
+            GraphQLInterfaceType.class,
+            GraphQLUnionType.class,
+            GraphQLObjectType.class,
+            GraphQLEnumType.class,
+            GraphQLScalarType.class,
+            GraphQLInputObjectType.class).flatMap(type -> printType(typesAsList, type, visibility)));
+
```

Comment:
This concat, stream / flatmap has 7 lines an of code and the procedural one above has 6 and is way easier to understand.

Why change it other then just preferring it this way?